### ISSUE TITLE
Chartjs-plugin-zoom not working with multiple charts?

### ISSUE THREAD

**Issue by darshg9** [chartjs/chartjs-plugin-zoom#58]
Hello,

I'm currently working on a dashboard using chartjs. I have several charts displayed on the same page and the zoom plugin applies only to one of it. Is multiple chart support not implemented yet or did I do something wrong?

Thanks for your future answer.

**Comment by etimberg** [comment]
@darshg9 do you have a link to a page which shows this behaviour so that we can debug?

**Comment by darshg9** [comment]
Unfortunately I'm in the middle of a vocational training (I'm a beginner developer) and this is a protected project for the society in wich I'm making an internship, so I can't give you a link. I can make a screenshot after hiding sensitive datas, though. And of course answer to your question as best as I can.

**Comment by darshg9** [comment]
Here are some screenshots.

![capture d ecran 2016-12-13 a 14 47 17](https://cloud.githubusercontent.com/assets/18553795/21143139/cbc371ce-c145-11e6-81ec-0b3d385e7b6d.png)
![capture d ecran 2016-12-13 a 14 48 19](https://cloud.githubusercontent.com/assets/18553795/21143138/cbc14a0c-c145-11e6-987e-9fd96488ad36.png)

The page contains several small cards with "quick view" charts (the background), each wich can be clicked to open the pop-up wich contains even more charts organized in two tabs. The zoom plugin applies only to the first of the two bubble charts wich are in the second tab. I think it's trying to apply to the other charts, though, since the page scrolling is "disabled" when hovering any of it.
What bugs me most is that the chart wich works isn't the first one of the page nor the last one, wich could makes sense. At the moment, I have no idea why the working one is this one and not any of the others.

**Comment by compwright** [comment]
I have multiple charts working with zoom on the same page, so I'm guessing there's something wrong with your setup. Check the browser console for Javascript errors, double-check (triple-check if you have to) your syntax, and setting up a jsfiddle or codepen is a good way to isolate where the problem is. If you can get that working, then you can systematically start comparing the differences to find the problem, or if you can't get it working, we'll have something to work with to identify the bug.

Also note that your chart types may be affecting the zoom plugin somehow, try changing them all to line charts and see what happens.

**Comment by darshg9** [comment]
As I said, I'm still a beginner so I expected it to probably has to do with an error I made.
I'll first try to change all my charts to line charts and then eventually try to reproduce my issue in a jsfiddle/codepen.

**Comment by darshg9** [comment]
Since converting my bubble charts to line charts implied to completely redefine their datasets structure, I tried to simply temporarily disable them (and all references to them, canvases included) as comments and switch all the other charts to line type. Now, all the remaining charts are line types and the zoom applies to none of them.
I'll try to reproduce all this in a jsfiddle/codepen.

**Comment by darshg9** [comment]
Ok, I finally found the the origin of my issue: I'm a retard. After wasting a day and a half arranging my code to make a codepen I could show you, I realized I simply added the options associated to the zoom plugin to *only one* of my charts when I was sure did it for each one... I knew I was tired, but didn't thought it was this bad. After two and a half month of good work (as my boss said), making such a stupid mistake just before the end of my internship makes my quite mad...

In conclusion, there is no problem with multiple charts at all, sorry for wasting your time.

**Comment by compwright** [comment]
Don't be so hard on yourself, it happens to the best of us, I promise :) Glad you found the problem.
